### PR TITLE
build: OwlBot configuration fix for monorepo

### PR DESCRIPTION
**Pull request by suztomo**
Preparing OwlBot configuration files for this gapic-generator-java monorepo.

### CODE REVIEW

File: java-iam/.OwlBot.yaml [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
-  - "/grpc-google-.*/src"
-  - "/proto-google-.*/src"
+  - "/java-iam/grpc-google-.*/src"
+  - "/java-iam/proto-google-.*/src"
 
 deep-preserve-regex:
-  - "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
-  - "/proto-google-iam-v1/src/main/java/com/google/iam/v1/logging"
-  - "/proto-google-iam-v1/src/main/proto/google/iam/v1"
+  - "/java-iam/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
```
Comment (blakeli0) [inline java-iam/.OwlBot.yaml:21]: Do we need to preserve this `google-` folder?

Comment (suztomo) [reply java-iam/.OwlBot.yaml:21]: No. Removing this line.

Comment (suztomo) [reply java-iam/.OwlBot.yaml]: Removed.

**Comment by suztomo** [comment]
The OwlBot Post Processor considers this repository as monorepo correctly. Good:

```
Step #5: gcr.io/cloud-devrel-public-resources/owlbot-java:latest@sha256:a57d2ea6d1a77aa96c17ad0850b779ec6295f88b6c1da3d214b2095d140a2066
Step #5: Processing monorepo
Step #5: In monorepo but no owl-bot-staging. Formatting changes in the last commit
```

https://pantheon.corp.google.com/cloud-build/builds;region=global/fd5ec261-4dfe-487d-93fc-d1b85c5c6071?project=repo-automation-bots&jsmode=O&mods=logs_tg_staging

**Review by ddixit14 — APPROVED** [review]

**Pull request merged**